gluster-exporter/metric_brick.go

```diff
@@ -188,6 +308,75 @@ func getGlusterSubvolLabels(volname string, subvol string) prometheus.Labels {
 	}
 }
 
+func getBrickDiskLabels(
+	brick glusterutils.Brick,
+	subvol string,
+	volume string,
+	device string,
+	disk string) prometheus.Labels {
+	return prometheus.Labels{
+		"host":       brick.Host,
+		"id":         brick.ID,
+		"brick_path": brick.Path,
+		"volume":     volume,
+		"subvolume":  subvol,
+		"device":     device,
+		"disk":       disk,
+	}
+}
+
+func getBrickDevice(brickPath string) string {
+	cmd := fmt.Sprintf("df --output=source %s", brickPath)
+	out, err := exec.Command("sh", "-c", cmd).Output()
+	if err != nil {
+		log.WithError(err).WithFields(log.Fields{
+			"command": cmd,
+		}).Error("Error executing command")
+		return ""
+	}
+	dfData := strings.Split(strings.TrimSpace(string(out)), "\n")[1]
+	fInfo, err := os.Lstat(dfData)
+	if err != nil {
+		log.WithError(err).WithFields(log.Fields{
+			"brickpath": brickPath,
+		}).Error("Error getting df data")
+		return ""
+	}
+	return fInfo.Name()
+}
+
+func getPhysicalDisk(blockDevice string) string {
+	cmd := "lsblk --all --bytes --raw --noheadings --output NAME,PKNAME,TYPE"
+	out, err := exec.Command("sh", "-c", cmd).Output()
+	if err != nil {
+		log.WithError(err).WithFields(log.Fields{
+			"command": cmd,
+		}).Error("Error executing command")
+		return ""
+	}
+	parentDetails := make(map[string]string)
+	lines := strings.Split(string(out), "\n")
+	for _, line := range lines {
+		if line != "" {
+			fields := strings.Split(line, " ")
+			if fields[1] != "" {
+				parentDetails[fields[0]] = fields[1]
```

Comment:
A logical volume can have multiple parent physical devices. In this case the last parent device iterated is the one that's getting mapped to the logical volume(previous ones get overwritten). So I would suggest defining parentDetails as `parentDetails := make( map[string][]string )` and then inside the for loop we have 
```
if fields[1] != "" {                                                                                                                                                           
     parentDetails[fields[0]] = append( parentDetails[fields[0]], fields[1] ) 
}  
```